galaxy_ng/tests/unit/api/test_api_v3_collections.py

```diff
@@ -127,8 +188,37 @@ def test_collection_versions_list(self):
         self.assertEqual(response.data['data'][0]['version'], '1.1.2')
         self.assertEqual(response.data['data'][1]['version'], '1.1.1')
 
+        # Ensure hrefs are overwritten
+        self.assertNotIn(self.pulp_href_fragment, response.data["data"][0]["href"])
+
         # TODO: implement subtests for each version after the
         # upload of artifacts has been implemented in `self.setUp`
         # for version in response.data['data']:
         #     with self.subTest(version=version['version):
         #         vresponse = self.client.get(version['href'])
+        #         self.assertNotIn(self.pulp_href_fragment, vresponse.data["href"])
+        #         self.assertNotIn(self.pulp_href_fragment, vresponse.data["collection"]["href"])
+        #         self.assertNotIn(self.pulp_href_fragment, vresponse.data["download_url"])
+
+    # def test_unpaginated_collection_versions_list(self):
+    #     """Assert the call to v3/collections/all returns correct
+    #     collections and versions
+    #     """
+    #     self.client.force_authenticate(user=self.admin_user)
+    #     response = self.client.get(self.all_versions_url)
+    #     self.assertEqual(response.data[0]['version'], '1.1.2')
+
+    #     # Ensure href is overwritten
+    #     self.assertNotIn(self.pulp_href_fragment, response.data[0]["href"])
+    #     self.assertNotIn(self.pulp_href_fragment, response.data[0]["collection"]["href"])
+    #     self.assertNotIn(self.pulp_href_fragment, response.data[0]["download_url"])
+
+    #     # Ensure some fields are in
+    #     for field in ('metadata', 'namespace', 'name', 'artifact'):
+    #         with self.subTest(field=field):
+    #             self.assertIn(field, response.data[0])
+
+    #     # Ensure some fields are not in
+    #     for field in ('manifest', 'files'):
+    #         with self.subTest(field=field):
+    #             self.assertNotIn(field, response.data[0])
```

Comment:
**NOTE** to enable those tests we have to include `assets/` and import a real collection, or create a mocked `ContentArtifact`. This probably will be part of a funtional `sync` test case later, left here as reference.